ruby/src/main/java/com/google/protobuf/jruby/RubyMessage.java

```diff
@@ -635,12 +664,8 @@ public IRubyObject toHash(ThreadContext context) {
         return ret;
     }
 
-    protected DynamicMessage build(ThreadContext context) {
-        return build(context, 0);
-    }
-
-    protected DynamicMessage build(ThreadContext context, int depth) {
-        if (depth > SINK_MAXIMUM_NESTING) {
+    protected DynamicMessage build(ThreadContext context, int depth, int maxRecursionDepth) {
+        if (depth >= maxRecursionDepth) {
```

Comment:
Note this is changed to match the UPB library behavior, and SINK_MAXIMUM_NESTING is incremented by one accordingly.